app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -101,6 +102,60 @@ protected void onCreate(Bundle savedInstanceState) {
         initBroadcastReceivers();
 
         mediator = new MainMediator(this);
+        permissionHandler = new PermissionHandler(new PermissionHandle() {
+            @Override
+            public void doAction(String permission, int actionId, int triggerType, Serializable params) {
+                if (actionId == ACTION_CAPTURE) {
+                    BrowserFragment browserFragment = getBrowserFragment();
+                    switch (triggerType) {
```

Comment:
Maybe check browserFragment null and early return before entering switch case?

Reply:
I don't assume every new action in the future will need a BrowserFragment.